encoder/libav_encoder.cpp

```diff
@@ -26,14 +28,41 @@ void encoderOptionsH264M2M(VideoOptions const *options, AVCodecContext *codec)
 		codec->bit_rate = options->bitrate;
 }
 
-static const std::map<std::string, std::function<void(VideoOptions const *, AVCodecContext *)>> optionsMap =
+void encoderOptionsLibx264(VideoOptions const *options, AVCodecContext *codec)
+{
+	codec->pix_fmt = AV_PIX_FMT_YUV420P;
+
+	if (options->bitrate)
+		codec->bit_rate = options->bitrate * 1000;
+
+	codec->max_b_frames = 0;
+	codec->me_range = 16;
+	codec->me_cmp = 1; // No chroma ME
+	codec->me_subpel_quality = 0;
+	codec->thread_count = 16;
+	codec->thread_type = FF_THREAD_FRAME;
+	codec->slices = 1;
+
+	av_opt_set(codec->priv_data, "preset", "superfast", 0);
+	av_opt_set(codec->priv_data, "partitions", "i8x8,i4x4", 0);
+	av_opt_set(codec->priv_data, "weightp", "none", 0);
+	av_opt_set(codec->priv_data, "motion-est", "dia", 0);
+	av_opt_set(codec->priv_data, "sc_threshold", "0", 0);
+	av_opt_set(codec->priv_data, "rc-lookahead", "0", 0);
+	av_opt_set(codec->priv_data, "mixed_ref", "0", 0);
+}
+
+const std::map<std::string, std::function<void(VideoOptions const *, AVCodecContext *)>> optionsMap =
 {
 	{ "h264_v4l2m2m", encoderOptionsH264M2M },
+	{ "libx264", encoderOptionsLibx264 },
 };
 
+} // namespace
+
 void LibAvEncoder::initVideoCodec(VideoOptions const *options, StreamInfo const &info)
 {
-	const std::string codec_name("h264_v4l2m2m");
+	const std::string codec_name(options->libav_video_codec);
```

Comment:
Just wondering very slightly whether the `codec_name` variable is even necessary? Not that it matters...

Reply:
I can remove that.